### PR TITLE
WT-7126 Coverity analysis defect 116991: Explicit null dereferenced

### PR DESCRIPTION
**Pull request by sueloverso**

### CODE REVIEW

**Comment by sueloverso** [comment]
evergreen retry

File: src/utilities/util_list.c [hunk 1]
```diff
@@ -88,7 +88,8 @@ list_init_block(WT_SESSION *session, const char *key, WT_BLOCK *block)
     wt_api = session->connection->get_extension_api(session->connection);
     if ((ret = wt_api->metadata_search(wt_api, session, key, &config)) != 0)
         WT_ERR(util_err(session, ret, "%s: WT_EXTENSION_API.metadata_search", key));
-    if ((ret = wt_api->config_parser_open(wt_api, session, config, strlen(config), &parser)) != 0)
+    if (config != NULL &&
+      (ret = wt_api->config_parser_open(wt_api, session, config, strlen(config), &parser)) != 0)
         WT_ERR(util_err(session, ret, "WT_EXTENSION_API.config_parser_open"));
     if ((ret = parser->get(parser, "allocation_size", &cval)) == 0)
```
Comment (keitharnoldsmith) [inline src/utilities/util_list.c:94]: If `config == NULL` then we'll wind up here with `parser == NULL` and try to dereference it.

**Review by keitharnoldsmith — CHANGES_REQUESTED** [review]

File: src/utilities/util_list.c (hunk 1 above)
Comment (sueloverso) [reply src/utilities/util_list.c:99]: Per our discussion, I went with a different approach.

**Review by keitharnoldsmith — APPROVED** [review]
LGTM

**Review by ddanderson — APPROVED** [review]
LGTM.

**Pull request merged**